scipy/ndimage/tests/test_morphology.py

```diff
@@ -2257,6 +2257,55 @@ def test_binary_fill_holes03(self, xp):
         out = ndimage.binary_fill_holes(data)
         assert_array_almost_equal(out, expected)
 
+    @skip_xp_backends("cupy",
+                      reasons=["these filters do not yet have axes support"])
+    @pytest.mark.parametrize('border_value',[0, 1])
+    @pytest.mark.parametrize('origin', [(0, 0), (-1, 0)])
+    @pytest.mark.parametrize('expand_axis', [0, 1, 2])
+    @pytest.mark.parametrize('func', [ndimage.binary_erosion,
+                                      ndimage.binary_dilation,
+                                      ndimage.binary_opening,
+                                      ndimage.binary_closing,
+                                      ndimage.binary_hit_or_miss,
+                                      ndimage.binary_propagation,
+                                      ndimage.binary_fill_holes])
+    def test_binary_axes(self, xp, func, expand_axis, origin, border_value):
+        struct = [[0, 1, 0],
+                  [1, 1, 1],
+                  [0, 1, 0]]
+
+        data = xp.array([[0, 0, 0, 1, 0, 0, 0],
```

Comment:
`xp.asarray` is in the standard, `xp.array` is not

Reply:
thanks, updated